Merge IK-92-2.0 into dev

adrielFab opened the pull request:
Feature: color code matching with driver route and passenger routes

adrielFab committed:
IK-93 This branch is inspired by IK-92. The implementation for the matching algorithm is complete

The first branch had the algorithm for matching points together. The point of this second branch was to implement the user interface aspect and refactoring the code to use the algorithm

adrielFab committed:
IK-93 Refactored the matching algorithm to actually include the end location of the route of the passenger

anrchen commented:
To be done in Testing concerning this feature:

- [x] 1) Test: create a match-able route 

```
Expectations: pink color code is displayed to confirm the match
Result: Passed
```

- [x] 2) Test: create a unmatch-able route 

```
Expectations: The route is created without any pink color code 
Result: Passed
```

- [x] 3) Test: create a unmatch-able route with the same start location 
```
Expectations: The route is created without any pink color code 
Result: Passed

```

- [x] 4) Test: create a unmatch-able route with the same end location 
```
Expectations: The route is created without any pink color code 
Result: Passed
```

anrchen commented on client/app/src/main/java/com/example/mrides/Activity/CreateRouteActivity.java:
```diff
@@ -217,31 +221,34 @@ public void onProviderDisabled(String s) {
-        BitmapDrawable bitmapdraw=(BitmapDrawable)getResources().getDrawable(R.drawable.passenger_icon);
-        Bitmap b=bitmapdraw.getBitmap();
-        Bitmap smallMarker = Bitmap.createScaledBitmap(b, width, height, false);
+//        int height = 100;
```
Remove unwanted comments from line 233 to 237

anrchen commented on client/app/src/main/java/com/example/mrides/Activity/CreateRouteActivity.java:
```diff
@@ -376,4 +386,75 @@ public void responseReceived(String response) {
+
+                while (i < routeOfUser.size() && pickUpBool == false) {
+                    LatLng pointInPoly = routeOfUser.get(i);
+//                    System.out.println("Distance " + (distance(pointInPoly.latitude, pointInPoly.longitude
```
Remove unwanted comments from line 404 to 405.

anrchen commented on client/app/src/main/java/com/example/mrides/Activity/CreateRouteActivity.java:
```diff
@@ -9,6 +9,7 @@
 import android.app.Dialog;
 import android.app.ProgressDialog;
 import android.content.Context;
+import android.content.SyncStatusObserver;
```
Unused import statement

anrchen commented on client/app/src/main/java/com/example/mrides/Activity/CreateRouteActivity.java:
```diff
@@ -9,6 +9,7 @@
 import android.content.Context;
+import android.content.SyncStatusObserver;
 import android.content.pm.PackageManager;
 import android.graphics.Bitmap;
```
Unused import statement on line 14 and 16 for `import android.graphics.drawable.BitmapDrawable; ` and `import android.graphics.Bitmap;`

anrchen commented on client/app/src/main/java/com/example/mrides/Activity/CreateRouteActivity.java:
```diff
@@ -376,4 +386,75 @@ public void responseReceived(String response) {
+        return dist; // output distance, in MILES
+    }
+
+    public boolean validateDistance(LatLng passengerLocation, LatLng userLocation) {
```
Add comment to explain what does "0.1" signify in the matching algorithm. For example: distance smaller than 160 meters.

anrchen reviewed (commented):
General modifications are needed in term of unwanted comments and unused imports.

anrchen committed:
IK-94 Unit tests implemented for color code of passengers

The following unit tests have been added: testShouldCreateProgressBar, testMatchRoute, testValideDistance, TestOnCancelListener

anrchen committed:
IK-63 View route unit test added

Unit test asserts that the route creation
